lib/common/http1client.c

```diff
@@ -468,15 +468,6 @@ static void on_whole_request_sent(h2o_socket_t *sock, const char *err)
         return;
     }
 
-    if (client->_is_chunked) {
-        client->_is_chunked = 0;
-        h2o_iovec_t last = h2o_iovec_init(H2O_STRLIT("0\r\n\r\n"));
-        client->super.bytes_written.body += last.len;
-        client->super.bytes_written.total += last.len;
-        h2o_socket_write(client->sock, &last, 1, on_whole_request_sent);
-        return;
-    }
```

Comment:
Fix in bad0ffe is half broken.

It is true that moving the emission of `0\r\n\r\n` to `encode_chunk`do_write_req` prevents `0\r\n\r\n` being sent twice at the end of the stream, if the last chunk (0 bytes of data + is_end_stream) is submitted while write is _not_ in flight.

But if write is inflight, with the changes proposed in bad0ffe, `\0\r\n\r\n` will not be sent, because`_body_buf_is_done` gets set while write is inflight. That is why CI is failing.

Reply:
Fixed in 2171851.